database/src/service/free.rs

```diff
@@ -0,0 +1,137 @@
+//! General free functions used (related to `cuprate_database::service`).
+
+//---------------------------------------------------------------------------------------------------- Import
+use std::sync::OnceLock;
+
+use crate::{
+    service::read::DatabaseReader,
+    service::write::DatabaseWriter,
+    service::{DatabaseReadHandle, DatabaseWriteHandle},
+    ConcreteEnv,
+};
+
+//---------------------------------------------------------------------------------------------------- const/static
+/// Read/write handles to the single, global program database.
+///
+/// [`init()`] will initialize this [`OnceLock`], and store
+/// the initialized database's handles inside it.
+///
+/// Not accessible publically, outside crates use one of:
+/// - [`init()`]
+/// - [`db_read()`]
+/// - [`db_write()`]
+static DATABASE_HANDLES: OnceLock<(DatabaseReadHandle, DatabaseWriteHandle)> = OnceLock::new();
+
+//---------------------------------------------------------------------------------------------------- Init
+#[cold]
+#[inline(never)] // Only called once.
+/// Initialize the database, the thread-pool, and return a read/write handle to it.
+///
+/// This initializes a [`OnceLock`] containing the [`ConcreteEnv`],
+/// meaning there is only 1 database per program.
+///
+/// Calling this function again will return handles to the same database.
+pub fn init() -> &'static (DatabaseReadHandle, DatabaseWriteHandle) {
```

Comment:
IMO I think it would be better to have the function `init` just spawn the DB and give handles back to the caller who can then give handles to the DB to whatever parts of Cuprate need it, instead of allowing any part of Cuprate to call `db_read`/ `db_write` and get a handle